### ISSUE TITLE
Bug: Duplicated autoclass directive warning breaks doc build

### ISSUE THREAD

**Issue by pushfoo** [pythonarcade/arcade#1891]
## Bug Report

tl;dr
1. A custom ClassDocumenter was added in cadf05e6
2. Sphinx API doesn't provide a clear way to replace a doc builder rather than add it
3. The resulting duplicate autoclass directive warning is promoted to an error, breaking our build

## System Info

arcade installed by branching off the tip of development ([908664e](https://github.com/pythonarcade/arcade/commit/908664efc256697d3098a347f63d217d97841782))

### Actual behavior:

Running doc build on CI or locally produces the following error:
 ```
WARNING: while setting up extension conf.py: directive 'autoclass' is already registered, it will be overridden
```
(See the [doc build](https://github.com/pythonarcade/arcade/actions/runs/6107475219/job/16574549730?pr=1890))

### Expected behavior:
No error.

### Steps to reproduce/example code:

Run CI, or install from tip of development and perform the following steps:

1. `pip install -I -e .[dev]`
2. `./make.py clean`
3. `./make.py html`
